### DISCUSSION TITLE
How to obtain the parameters of a component？

### DISCUSSION THREAD

**Discussion by RickYang1** [InfinyTech3D/SofaUnity#106] [Help - Using SofaUnity]
I want to obtain the force and constraint arrays in MechanicalObject, as well as identify the objects and their elements that are in contact. I haven’t found a usable method for this. How can I do this?
Thanks in advance.

**Comment by epernod** [comment]
Hi,

when you say the objects and their elements that are in contact. You mean the name of the CollisionModel ? and the index of the element with this Model? 
For the force and constraint array if should be possible using the DataArray API. I will need to test on my side.

Best regards,

**Reply by RickYang1** [reply]
> Hi, 你好，
> 
> when you say the objects and their elements that are in contact. You mean the name of the CollisionModel ? and the index of the element with this Model?当你说物体及其接触的元素时。您是指 CollisionModel 的名称吗？以及该模型元素的索引？ For the force and constraint array if should be possible using the DataArray API. I will need to test on my side.对于力和约束数组，应该可以使用 DataArray API。我需要在我这边进行测试。
> 
> Best regards, 此致，

Hello,
Yes,what I want to obtain is the name of CollisionModel and the index.

**Reply by epernod** [reply]
Hi,

do you still need this?
Best regards,
